sdk/servicebus/azure-messaging-servicebus-track2-perf/src/main/java/com/azure/messaging/servicebus/perf/core/ServiceTest.java

```
@@ -0,0 +1,85 @@
+ */
+public abstract class ServiceTest<TOptions extends PerfStressOptions> extends PerfStressTest<TOptions> {
+    private final ClientLogger logger = new ClientLogger(ServiceTest.class);
+    protected static final Duration TIMEOUT = Duration.ofSeconds(60);
+    protected static final AmqpRetryOptions RETRY_OPTIONS = new AmqpRetryOptions().setTryTimeout(TIMEOUT);
+    protected static final String CONTENTS = "Track 2 AMQP message - Perf Test";
+    protected static final int TOTAL_MESSAGE_MULTIPLIER = 300;
+
+    private static final String AZURE_SERVICE_BUS_CONNECTION_STRING = "AZURE_SERVICE_BUS_CONNECTION_STRING";
+    private static final String AZURE_SERVICEBUS_QUEUE_NAME = "AZURE_SERVICEBUS_QUEUE_NAME";
+    private static final String AZURE_SERVICEBUS_TOPIC_NAME = "AZURE_SERVICEBUS_TOPIC_NAME";
+    private static final String AZURE_SERVICEBUS_SUBSCRIPTION_NAME = "AZURE_SERVICEBUS_SUBSCRIPTION_NAME";
+
+    protected final ServiceBusReceiverClient receiver;
+    protected final ServiceBusReceiverAsyncClient receiverAsync;
+    protected final ServiceBusSenderClient sender;
+    protected final ServiceBusSenderAsyncClient senderAsync;
+
+    /**
+     *
+     * @param options to configure.
+     * @param receiveMode to receive messages.
+     * @throws IllegalArgumentException if environment variable not being available.
+     */
+    public ServiceTest(TOptions options, ReceiveMode receiveMode) {
+        super(options);
+        String connectionString = System.getenv(AZURE_SERVICE_BUS_CONNECTION_STRING);
+        if (CoreUtils.isNullOrEmpty(connectionString)) {
+            logger.logExceptionAsError(new IllegalArgumentException("Environment variable "
+                + AZURE_SERVICE_BUS_CONNECTION_STRING + " must be set."));
+        }
+
+        String queueName = System.getenv(AZURE_SERVICEBUS_QUEUE_NAME);
+        if (CoreUtils.isNullOrEmpty(queueName)) {
+            logger.logExceptionAsError(new IllegalArgumentException("Environment variable "
```

you need to throw these errors.